frontend/src/libs/packages/storage/storage.package.ts

```diff
@@ -1,35 +1,73 @@
-import { type ValueOf } from '~/libs/types/types.js';
+import { ApplicationError } from 'shared/build/index.js';
 
 import { type StorageKey } from './libs/enums/enums.js';
 import { type IStorage } from './libs/interfaces/interfaces.js';
 
+type StorageKeyValue = keyof typeof StorageKey;
+
 class Storage implements IStorage {
   private store: globalThis.Storage;
 
   public constructor(store: globalThis.Storage) {
     this.store = store;
   }
 
-  public set(key: ValueOf<typeof StorageKey>, value: string): Promise<void> {
-    this.store.setItem(key as string, value);
+  public set<T>(key: StorageKeyValue, value: T): Promise<void> {
+    try {
+      const serializedValue = JSON.stringify(value);
+      this.store.setItem(key as string, serializedValue);
 
-    return Promise.resolve();
+      return Promise.resolve();
+    } catch (error) {
+      return Promise.reject(
+        new ApplicationError({
+          message: `Error setting value in storage '${key}'`,
+          cause: error,
+        }),
+      );
+    }
   }
 
-  public get<R = string>(key: ValueOf<typeof StorageKey>): Promise<R | null> {
-    return Promise.resolve(this.store.getItem(key as string) as R);
+  public get<T>(key: StorageKeyValue): Promise<T | null> {
+    try {
+      const value = this.store.getItem(key as string);
+
+      return value === null
+        ? Promise.resolve(null)
+        : Promise.resolve(JSON.parse(value) as T);
```

Comment:
```suggestion
      return Promise.resolve(JSON.parse(value) as T)
```
JSON.parse(null) will return null 🤔